extensions/mssql/src/tableDesigner/tableDesigner.ts

```diff
@@ -0,0 +1,31 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { AppContext } from '../appContext';
+import * as azdata from 'azdata';
+import * as vscode from 'vscode';
+import { sqlProviderName } from '../constants';
+
+export function registerTableDesignerCommands(appContext: AppContext) {
+	appContext.extensionContext.subscriptions.push(vscode.commands.registerCommand('mssql.newTable', async (context: azdata.ObjectExplorerContext) => {
+		await azdata.designers.openTableDesigner(sqlProviderName, {
+			server: context.connectionProfile.serverName,
+			database: context.connectionProfile.databaseName,
+			isNewTable: true
+		});
+	}));
+
+	appContext.extensionContext.subscriptions.push(vscode.commands.registerCommand('mssql.designTable', async (context: azdata.ObjectExplorerContext) => {
+		const nameParts = context.nodeInfo.label.split('.');
```

Comment:
The nodeInfo has `metadata.schema` and `metadata.name` already - which should be used since splitting like this probably isn't going to work if the name has a period in it already.

Reply:
good point.